src/constraints/joints/mod.rs

```diff
@@ -96,10 +96,10 @@ pub use revolute::*;
 pub use spherical::*;
 
 use crate::prelude::*;
-use bevy::prelude::*;
+use bevy::{ecs::entity::MapEntities, prelude::*};
 
 /// A trait for [joints].
-pub trait Joint: Component + PositionConstraint + AngularConstraint {
```

Comment:
If we want to add `MapEntities` to the trait bounds, would it make sense to have it on `XpbdConstraint` directly? This way it'd be required for all constraints and not just joints, and all constraints currently have entities anyway